File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1
        while (l < r):
            c = numbers[l] + numbers[r]
            if  numbers[l] + numbers[r] == target:
                return [l + 1, r + 1]
            elif c < target:
                l += 1
            else:
                r -= 1

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, r = nums.length - 1;
        // find the pivot index
        while (l < r) {
            int m = l + (r - l) / 2;
            if (nums[m] > nums[r]) {
                l = m + 1;
            } else {
                r = m;
            }
        }
        int pivot = l;
        l = 0;
        r = nums.length - 1;
        if (target >= nums[pivot] && target <= nums[r]) {
            l = pivot;
        } else {
            r = pivot - 1;
        }
        while (l <= r) {
            int m = l + (r-l)/2;
            if (nums[m] == target ) return m;
            else if (target > nums[m]) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }

        return -1;
    }
}
